test/common/run-tests

```diff
@@ -65,14 +65,24 @@ def print_test(test, print_tap=True):
         print("not ok " + test_name(test))
     sys.stdout.flush()
 
-def finish_test(opts, test):
+def finish_test(opts, test, affected_tests):
     """Returns if a test should retry or not
 
     Call test-policy on the test's output, print if needed.
 
     Return (retry_reason, exit_code). retry_reason can be None or a string.
     """
-    if test.process.returncode in [0, 77]:
+
+    affected = len(affected_tests) < 3 and test.command[0].encode("utf-8") in affected_tests
```

Comment:
The "< 3" is inconsistent with the "more than 3" in the commit message. Also, I find it a bit inelegant to always re-do both this length test and the decoding here -- could they be done just once, right wher the `affected_tests` list is built below? Then this should also  get a comment "why 3?"